### PR TITLE
Jitclass static methods

### PR DESCRIPTION
**Pull request by EPronovost**
Addresses https://github.com/numba/numba/issues/4451

The following works now
```
        @jitclass(spec=[("x", int32)])
        class Test:
            def __init__(self, x):
                self.x = x

            def increase(self, y):
                self.x = self.add(self.x, y)
                return self.x

            @staticmethod
            def add(a, b):
                return a + b

        t = Test(0)
        self.assertEqual(1, t.increase(1))
        self.assertEqual(3, Test.add(1, 2))
```
Note that replacing `self.add` with `Test.add` in `increase()` fails with `Untyped global name 'Test': cannot determine Numba type of <class 'type'>`.  I'm not sure if this can be resolved easily.

### CODE REVIEW

**Comment by gmarkall** [comment]
Thanks for the PR! Marking as ready for review.

One observation I have is that the tests only cover the addition of a single static method to a single JIT class - is it possible to add tests of multiple static methods to multiple JIT classes, with different signatures? I haven't looking in detail at the implementation (and I'm not familiar with the jitclass implementation in general) but I think that covering a few more cases would help towards the robustness of the implementation in future.

**Comment by gmarkall** [comment]
Maybe someone else can comment on:

> Note that replacing `self.add` with `Test.add` in `increase()` fails with `Untyped global name 'Test': cannot determine Numba type of <class 'type'>`. I'm not sure if this can be resolved easily.

I imagine there's a way around it, but I'd have to dig into the details of the implementation to work out what it might be.

**Comment by esc** [comment]
@EPronovost sure, I just added it to the milestone.

File: docs/source/user/jitclass.rst [hunk 1]
```diff
@@ -162,6 +168,9 @@ must be unboxed or boxed between Python objects and native representation.
 Values encapsulated by a jitclass does not get boxed into Python object when
 the jitclass instance is handed to the interpreter.  It is during attribute
 access to the field values that they are boxed.
+Calling static methods as class attributes is only supported outside of the
+class definition (i.e. you can't call ``Bag.add()`` from within another method
+of ``Bag``).
```
Comment (gmarkall) [inline docs/source/user/jitclass.rst:173]: The error message when calling a static method as a class attribute from another method doesn't appear to be tested.

Comment (stuartarchibald) [reply docs/source/user/jitclass.rst:148]: Should be done in 2acbd48

File: numba/core/types/misc.py [hunk 2]
```diff
@@ -410,8 +418,11 @@ def get_call_signatures(self):
 
     @property
     def methods(self):
-        methods = dict((k, v.py_func) for k, v in self.jitmethods.items())
-        return methods
+        return dict((k, v.py_func) for k, v in self.jit_methods.items())
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply numba/core/types/misc.py]: 2acbd48

File: docs/source/user/jitclass.rst [hunk 3]
```diff
@@ -45,9 +45,13 @@ Here's an example of a jitclass::
...
-              self.array[i] = val
+              self.array[i] = self.add(self.array[i], val)
           return self.array
 
+      @staticmethod
+      def add(x, y):
+          return x + y
+
 
 (see full example at `examples/jitclass.py` from the source tree)
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply docs/source/user/jitclass.rst]: a23594f

**Pull request merged**